### PR TITLE
Add MockTcgPpiLib

### PR DESCRIPTION
**Pull request by PaddyDengKC**
## Description

To provide google test mock of TcgPpi

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?
- [X] Backport to release branch?

## How This Was Tested

Tested on other consumer package that can leverage the MockTcgPpiLib for google test.

## Integration Instructions

N/A

### CODE REVIEW

File: SecurityPkg/Test/Mock/Include/GoogleTest/Ppi/MockTcgPpiLib.h [hunk 1]
```diff
@@ -0,0 +1,117 @@
+/** @file MockTcgPpiLib.h
+  Google Test mocks for TcgPpi
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#ifndef __MOCK_TCG_PPI_H__
```
Comment (apop5) [inline SecurityPkg/Test/Mock/Include/GoogleTest/Ppi/MockTcgPpiLib.h]:
```suggestion
#ifndef MOCK_TCG_PPI_H__

```

leading underscores are reserved for compiler use

**Comment by apop5** [comment]
Is there a corresponding PR into EDK2 to add this file there?

**Review by apop5 — APPROVED** [review]

**Review by VivianNK — APPROVED** [review]

**Comment by PaddyDengKC** [comment]
> Is there a corresponding PR into EDK2 to add this file there?

No, we only add this mock in here.

**Comment by VivianNK** [comment]
> > Is there a corresponding PR into EDK2 to add this file there?
> 
> No, we only add this mock in here.

For this change and all mocks moving forward, please create a PR to edk2 first.
[GoogleTest Guide](https://microsoft.sharepoint.com/:fl:/s/30977eb5-61b2-4736-846f-0ce26a79a787/EWH8IeVQPvxGornVBD0dGdABoJHByflI4MhDluxoz56Mzg?e=mZ5Rcz&nav=cz0lMkZzaXRlcyUyRjMwOTc3ZWI1LTYxYjItNDczNi04NDZmLTBjZTI2YTc5YTc4NyZkPWIlMjFyTDY1UHlPUG9rTzZLczRpTDNmMFRPZGpmb1FSa1g1S2plWF9zS2RETy1jc3RQTjFqZ1U5UnBhSHBnRzU2cDAzJmY9MDFLSkhRSktUQjdRUTZLVUI2N1JES0ZPT1ZBUTZSMkdPUSZjPSUyRiZhPUxvb3BBcHAmcD0lNDBmbHVpZHglMkZsb29wLXBhZ2UtY29udGFpbmVyJng9JTdCJTIydyUyMiUzQSUyMlQwUlRVSHh0YVdOeWIzTnZablF1YzJoaGNtVndiMmx1ZEM1amIyMThZaUZ5VERZMVVIbFBVRzlyVHpaTGN6UnBURE5tTUZSUFpHcG1iMUZTYTFnMVMycGxXRjl6UzJSRVR5MWpjM1JRVGpGcVoxVTVVbkJoU0hCblJ6VTJjREF6ZkRBeFMwcElVVXBMVWtWTlZrbEpNMHBCU1U5S1F6TkVRbGxVTlVGUk16TlFVbG8lM0QlMjIlMkMlMjJpJTIyJTNBJTIyYjNhOWMwNmMtMDIyOS00MmY3LTgxOTItM2RlZTAyZjRlNzIyJTIyJTdE)

**Comment by PaddyDengKC** [comment]
> > > Is there a corresponding PR into EDK2 to add this file there?
> > 
> > 
> > No, we only add this mock in here.
> 
> For this change and all mocks moving forward, please create a PR to edk2 first. [GoogleTest Guide](https://microsoft.sharepoint.com/:fl:/s/30977eb5-61b2-4736-846f-0ce26a79a787/EWH8IeVQPvxGornVBD0dGdABoJHByflI4MhDluxoz56Mzg?e=mZ5Rcz&nav=cz0lMkZzaXRlcyUyRjMwOTc3ZWI1LTYxYjItNDczNi04NDZmLTBjZTI2YTc5YTc4NyZkPWIlMjFyTDY1UHlPUG9rTzZLczRpTDNmMFRPZGpmb1FSa1g1S2plWF9zS2RETy1jc3RQTjFqZ1U5UnBhSHBnRzU2cDAzJmY9MDFLSkhRSktUQjdRUTZLVUI2N1JES0ZPT1ZBUTZSMkdPUSZjPSUyRiZhPUxvb3BBcHAmcD0lNDBmbHVpZHglMkZsb29wLXBhZ2UtY29udGFpbmVyJng9JTdCJTIydyUyMiUzQSUyMlQwUlRVSHh0YVdOeWIzTnZablF1YzJoaGNtVndiMmx1ZEM1amIyMThZaUZ5VERZMVVIbFBVRzlyVHpaTGN6UnBURE5tTUZSUFpHcG1iMUZTYTFnMVMycGxXRjl6UzJSRVR5MWpjM1JRVGpGcVoxVTVVbkJoU0hCblJ6VTJjREF6ZkRBeFMwcElVVXBMVWtWTlZrbEpNMHBCU1U5S1F6TkVRbGxVTlVGUk16TlFVbG8lM0QlMjIlMkMlMjJpJTIyJTNBJTIyYjNhOWMwNmMtMDIyOS00MmY3LTgxOTItM2RlZTAyZjRlNzIyJTIyJTdE)

Created https://github.com/tianocore/edk2/pull/6363.
Once the PR in the EDK2 repo completed I will update in this PR.

File: SecurityPkg/Test/Mock/Library/GoogleTest/Ppi/MockTcgPpiLib/MockTcgPpiLib.cpp [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/** @file MockTcgPpiLib.cpp
+  Google Test mocks for TcgPpi
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+#include <GoogleTest/Ppi/MockTcgPpiLib.h>
```
Comment (VivianNK) [inline SecurityPkg/Test/Mock/Library/GoogleTest/Ppi/MockTcgPpiLib/MockTcgPpiLib.cpp]: We actually don't need to have a cpp or inf file for a PPI mock. Please move the mock definitions to the header file.

Comment (PaddyDengKC) [reply SecurityPkg/Test/Mock/Library/GoogleTest/Ppi/MockTcgPpiLib/MockTcgPpiLib.cpp]: Hi Vivian,
I just pushed a commit to remove them. Please help to review it.

Thanks,
Paddy

File: SecurityPkg/Test/Mock/Include/GoogleTest/Ppi/MockTcgPpi.h [hunk 3]
```diff
@@ -0,0 +1,122 @@
+/** @file MockTcgPpiLib.h
+  Google Test mocks for TcgPpi
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#ifndef MOCK_TCG_PPI_H__
+
+#define __MOCK_TCG_PPI_H__
```
Comment (VivianNK) [inline SecurityPkg/Test/Mock/Include/GoogleTest/Ppi/MockTcgPpi.h]: Please change to match line #8

**Pull request merged**